src/runtime/runtime_rpi3.go

```diff
@@ -0,0 +1,140 @@
+// +build rpi3
+
+package runtime
+
+import (
+	"machine"
+	"runtime/volatile"
+	"unsafe"
+	"device/arm"
+)
+
+const tickMicros = int64(1)
+
+type timeUnit int64
+
+//go:export sleepticks sleepticks
+func sleepTicks(n timeUnit) {
+	WaitMuSec(uint32(n))
```

Comment:
You can handle longer durations by checking for overflow.
```suggestion
	for n > timeUnit(^uint32(0)) {
		WaitMuSec(^uint32(0))
		n -= timeUnit(^uint32(0))
	}
	WaitMuSec(uint32(n))
```

Reply:
Allowed 64bit wait values.